Data/NF/Internal.hs

```diff
@@ -17,11 +21,17 @@
 -- Stability   :  experimental
 -- Portability :  portable
 --
--- This module provides the constructor for the type 'NF', allowing
--- you to create values of type 'NF' without any runtime overhead
--- if you fulfill a proof obligation that the value is already in
--- normal form.
-module Data.NF.Internal(NF(..)) where
+-- This module is intended for internal use.
+module Data.NF.Internal(
+#if __GLASGOW_HASKELL__ >= 800
+  NF(..,NF)
+#else
+  NF(..)
+#if __GLASGOW_HASKELL__ >= 710
+  , pattern NF
```

Comment:
Ugh, it kind of sucks that we can't offer the NF constructor on earlier versions of Haskell. Well, I leave it to you.